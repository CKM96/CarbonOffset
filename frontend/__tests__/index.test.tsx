import { render } from '@testing-library/react';
import Home from '../pages/index';
import fetchMock from 'jest-fetch-mock';

jest.mock('jwt-decode', () => () => ({ sub: 'accountId' }));

beforeAll(() => {
  document.cookie = `accessToken=accountId;Max-Age=3600`;
  const projects = [
    {
      id: 'id',
      accountId: 'accountId',
      name: 'Project name',
      description: 'Project description',
    },
    {
      id: 'id2',
      accountId: 'accountId2',
      name: 'Second project name',
      description: 'Second project description',
    },
  ];
  fetchMock.mockIf('http://localhost:3001/projects', JSON.stringify(projects));
});

afterEach(jest.clearAllMocks);

describe('Home', () => {
  it('matches snapshot', async () => {
    const { container, findByText } = render(<Home />);

    await findByText('Project name');

    expect(container).toMatchSnapshot();
  });
});
