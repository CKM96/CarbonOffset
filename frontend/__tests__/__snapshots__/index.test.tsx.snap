// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ bWBldo"
  >
    <h1>
      Carbon Offsetting Projects
    </h1>
    <div
      class="sc-dkzDqf dJcJSQ"
    >
      <button>
        Create new project
      </button>
      <button>
        Logout
      </button>
    </div>
    <div
      class="sc-gsnTZi feoVmg"
    >
      <button>
        Edit
      </button>
      <h3>
        Project name
      </h3>
      <div>
        Project description
      </div>
    </div>
    <div
      class="sc-gsnTZi feoVmg"
    >
      <h3>
        Second project name
      </h3>
      <div>
        Second project description
      </div>
    </div>
  </div>
</div>
`;
